src/Contract/Transaction.purs

```diff
@@ -201,13 +211,173 @@ calculateMinFeeM
   :: forall (r :: Row Type). Transaction -> Contract r (Maybe Coin)
 calculateMinFeeM = map hush <<< calculateMinFee
 
+-- | Helper to adapt to UsedTxOuts
+withUsedTxouts
+  :: forall (r :: Row Type) (a :: Type)
+   . ReaderT UsedTxOuts (Contract r) a
+  -> Contract r a
+withUsedTxouts f = asks (_.usedTxOuts <<< unwrap) >>= runReaderT f
+
 -- | Attempts to balance an `UnattachedUnbalancedTx`.
 balanceTx
   :: forall (r :: Row Type)
    . UnattachedUnbalancedTx
   -> Contract r (Either BalanceTxError.BalanceTxError UnattachedTransaction)
 balanceTx = wrapContract <<< BalanceTx.balanceTx
 
+-- Helper to avoid repetition
+withTransactions
+  :: forall (a :: Type)
+       (t :: Type -> Type)
+       (r :: Row Type)
+       (tx :: Type)
+   . Traversable t
+  => (t UnattachedUnbalancedTx -> Contract r (t tx))
```

Comment:
Let's make it less polymorphic: we can just use Array. The downside of using APIs with constrained container types is that when you use something like `Map.toUnfoldable`, an extra type annotation is needed.

We will also get simpler type signatures which is a plus. In general, we aim for interface simplicity.

Reply:
I am using it with `Array.NonEmpty` from below...